api/graphql/types/reservation_unit/serializers.py

```diff
@@ -301,25 +315,13 @@ def update_hauki(instance: ReservationUnit) -> None:
     @staticmethod
     def handle_pricings(pricings: list[dict[Any, Any]], reservation_unit) -> None:
         # Delete pricings that are not in the payload
-        if not ReservationUnitPricingHelper.contains_status(PricingStatus.PRICING_STATUS_ACTIVE, pricings):
-            ReservationUnitPricing.objects.filter(
-                reservation_unit=reservation_unit,
-                status=PricingStatus.PRICING_STATUS_ACTIVE,
-            ).delete()
-
-        if not ReservationUnitPricingHelper.contains_status(PricingStatus.PRICING_STATUS_FUTURE, pricings):
-            ReservationUnitPricing.objects.filter(
-                reservation_unit=reservation_unit,
-                status=PricingStatus.PRICING_STATUS_FUTURE,
-            ).delete()
+        pricing_pks = [pricing.get("pk") for pricing in pricings if "pk" in pricing]
```

Comment:
Should probably add a transaction for the whole serializer save operation, so that this cannot succeed partially.

Reply:
Good catch!